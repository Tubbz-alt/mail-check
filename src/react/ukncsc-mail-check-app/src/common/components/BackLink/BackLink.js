import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import { Icon } from 'semantic-ui-react';

import './BackLink.css';

const DynamicBackLink = ({ link, history, children }) => (
  <p>
    {link ? (
      <Link to={link}>{children}</Link>
    ) : (
      <span className="BackLink" onClick={history.goBack}>
        {children}
      </span>
    )}
  </p>
);

const BackLink = props => (
  <DynamicBackLink {...props}>
    <Icon name="caret left" style={{ textDecoration: 'none' }} />Back
  </DynamicBackLink>
);

export default withRouter(BackLink);
