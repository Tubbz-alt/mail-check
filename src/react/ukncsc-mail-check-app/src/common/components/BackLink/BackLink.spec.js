import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from 'react-testing-library';
import BackLink from './BackLink';

describe('BackLink', () => {
  let container;

  describe('when rendered with a link', () => {
    beforeEach(() => {
      ({ container } = render(
        <MemoryRouter>
          <BackLink link="/bleh" />
        </MemoryRouter>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should pass the provided link to the Link component', () =>
      expect(container.querySelector('a').getAttribute('href')).toEqual(
        '/bleh'
      ));
  });

  describe('when rendered without a link', () => {
    beforeEach(() => {
      ({ container } = render(
        <MemoryRouter>
          <BackLink />
        </MemoryRouter>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });
});
