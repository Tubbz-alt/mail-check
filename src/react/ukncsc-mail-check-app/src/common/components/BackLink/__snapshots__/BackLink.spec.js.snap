// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackLink when rendered with a link it should match the snapshot 1`] = `
<div>
  <p>
    <a
      href="/bleh"
    >
      <i
        style="text-decoration: none;"
        aria-hidden="true"
        class="caret left icon"
      />
      Back
    </a>
  </p>
</div>
`;

exports[`BackLink when rendered without a link it should match the snapshot 1`] = `
<div>
  <p>
    <span
      class="BackLink"
    >
      <i
        style="text-decoration: none;"
        aria-hidden="true"
        class="caret left icon"
      />
      Back
    </span>
  </p>
</div>
`;
